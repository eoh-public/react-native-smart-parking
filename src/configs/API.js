import Config from 'react-native-config';

const API_ROOT = Config.API_ROOT;

const API = {
  AUTH: {
    LOGIN_PHONE: API_ROOT + '/accounts/login/',
    LOGIN_FACEBOOK: API_ROOT + '/accounts/login/facebook/',
    REGISTER_PHONE: API_ROOT + '/accounts/register/phone/',
    RESEND_OTP: API_ROOT + '/accounts/register/resend_otp/',
    VERIFY_OTP: API_ROOT + '/accounts/register/verify_otp/',
    LOGIN_SOCIAL_FB: API_ROOT + '/accounts/login/facebook/',
    LOGIN_SOCIAL_GG: API_ROOT + '/accounts/login/google/',
    LOGIN_SOCIAL_AP: API_ROOT + '/accounts/login/apple/',
    FORGOT_PASSWORD: API_ROOT + '/accounts/forgot_password/',
    FORGOT_PASSWORD_VERIFY_OTP:
      API_ROOT + '/accounts/forgot_password/verify_otp/',
    RESET_PASSWORD: API_ROOT + '/accounts/forgot_password/set_password/',
  },
  ACCOUNTS: {
    FEATURES: API_ROOT + '/accounts/features/',
    ADD_CARD: API_ROOT + '/billing/payments/stripe/add_card/',
    REMOVE_CARD: (id) =>
      API_ROOT + `/billing/payments/stripe/remove_cards/${id}/`,
    CHANGE_DEFAULT_CARD: API_ROOT + '/billing/payments/stripe/set_default/',
    LIST_PAYMENT_METHODS: API_ROOT + '/billing/list_payment_methods/',
    CREATE_CARD_TOKEN: 'https://api.stripe.com/v1/tokens',
  },
  UNIT: {
    ADD_USER: API_ROOT + '/property_manager/shared_units/add_by_phone_number/',
    MY_UNITS: API_ROOT + '/property_manager/units/mine/',
    SHARED_UNITS: API_ROOT + '/property_manager/shared_units/',
    UNIT_DETAIL: (id) => API_ROOT + `/property_manager/units/${id}/`,
    UNIT_NEAR_ME: (lat, lon, page) =>
      API_ROOT + `/property_manager/units/near-me/${lat},${lon}/?page=${page}`,
    UNITS_PUBLIC: API_ROOT + '/property_manager/units/public/',
    UNIT_SUMMARY: (id) => API_ROOT + `/property_manager/units/${id}/summary/`,
    UNIT_SUMMARY_DETAIL: (id, id2) =>
      API_ROOT + `/property_manager/units/${id}/summary_detail/${id2}/`,
    FILTER_SHARED_UNITS: (field) =>
      API_ROOT + `/property_manager/shared_units/?ordering=${field}`,
    STAR_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/star/`,
    UNSTAR_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/unstar/`,
    PIN_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/pin/`,
    UNPIN_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/unpin/`,
    MANAGE_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/`,
  },
  SUB_UNIT: {
    REMOVE_SUB_UNIT: (unitId, id) =>
      API_ROOT + `/property_manager/${unitId}/sub_units/${id}/`,
    CREATE_SUB_UNIT: (unitId) =>
      API_ROOT + `/property_manager/${unitId}/sub_units/`,
    MANAGE_SUB_UNIT: (unitId, id) =>
      API_ROOT + `/property_manager/${unitId}/sub_units/${id}/`,
    SENSOR_SCAN: (id) =>
      API_ROOT + `/property_manager/stations/${id}/sensor_scan/`,
  },
  SENSOR: {
    DISPLAY: (id) => API_ROOT + `/property_manager/sensors/${id}/display/`,
    DISPLAY_VALUES: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/display_values/`,
    DISPLAY_VALUES_V2: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/display_values_v2/`,
    DISPLAY_HISTORY: (id, hId) =>
      API_ROOT + `/property_manager/sensors/${id}/display_history/`,
    CAMERA_DISPLAY: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/cameras/`,
    REMOTE_CONTROL_OPTIONS: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/remote_control_options/`,
    QUICK_ACTION: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/quick_action/`,
    CHECK_CONNECTION: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/check_connection/`,
    UPDATE_SENSOR: (unit_id, station_id, id) =>
      API_ROOT +
      `/property_manager/${unit_id}/sub_units/${station_id}/devices/${id}/`,
  },
  CONFIG: {
    DISPLAY_HISTORY: API_ROOT + '/property_manager/configs/display_history/',
  },
  POWER_CONSUME: {
    DISPLAY_HISTORY:
      API_ROOT + '/property_manager/power_consume/display_history/',
  },
  SHARE: {
    UNITS: API_ROOT + '/property_manager/sharing/units/',
    UNIT_PERMISSIONS: (id) =>
      API_ROOT + `/property_manager/sharing/units/${id}/permissions/`,
    UNITS_MEMBERS: (id) =>
      API_ROOT + `/property_manager/sharing/units/${id}/members/`,
    UNITS_MEMBER_DETAIL: (id, shareId) =>
      API_ROOT + `/property_manager/sharing/units/${id}/members/${shareId}/`,
    SHARE: API_ROOT + '/property_manager/sharing/share/',
  },
  EXTERNAL: {
    WEATHER: (location, weatherMapId) =>
      `https://api.openweathermap.org/data/2.5/weather?lat=${location.lat}&lon=${location.lon}` +
      `&appid=${weatherMapId}`,
    UV_INDEX: (location, weatherMapId) =>
      `https://api.openweathermap.org/data/2.5/uvi?appid=${weatherMapId}&lat=${location.lat}&lon=${location.lon}`,
    AQI: (location, weatherMapToken) =>
      `https://api.waqi.info/feed/geo:${location.lat};${location.lon}/?token=${weatherMapToken}`,
    GOOGLE_MAP: {
      AUTO_COMPLETE:
        'https://maps.googleapis.com/maps/api/place/autocomplete/json',
      GET_LAT_LNG_BY_PLACE_ID:
        'https://maps.googleapis.com/maps/api/place/details/json',
    },
  },
  PARKING: {
    NEARBY: API_ROOT + '/smart_parking/parking_areas/near_me/',
    NEAREST:
      API_ROOT + '/smart_parking/parking_areas/nearest_available_parking/',
    SAVED_LIST: API_ROOT + '/smart_parking/parking_areas/saved_list/',
    SAVE: (id) => API_ROOT + `/smart_parking/parking_areas/${id}/save/`,
    UNSAVE: (id) => API_ROOT + `/smart_parking/parking_areas/${id}/unsave/`,
    DETAIL: (id) => API_ROOT + `/smart_parking/parking_areas/${id}/`,
    GET_BOOKING_PRICE: (id) =>
      API_ROOT + `/smart_parking/parking_areas/${id}/get_price/`,
    AVAILABLE_TIME_SLOTS: (id) =>
      API_ROOT + `/smart_parking/parkings/${id}/available-time-slots/`,
    PAYMENT_SUCCESS: (id) => API_ROOT + `/smart_parking/payment_success/${id}/`,
    CHECK_CAR_PARKED: API_ROOT + '/smart_parking/spots/check_car_parked/',
    PARKING_INFO: API_ROOT + '/smart_parking/spots/parking_info/',
  },
  CAR: {
    MY_CARS: API_ROOT + '/smart_parking/cars/',
    CHECK_CARS_INFO: API_ROOT + '/smart_parking/cars/check_information/',
    REMOVE_CAR: (id) => API_ROOT + `/smart_parking/cars/${id}/`,
    UPDATE: (id) => API_ROOT + `/smart_parking/cars/${id}/`,
    UPDATE_DEFAULT: (id) =>
      API_ROOT + `/smart_parking/cars/${id}/update_default/`,
  },
  BOOKING: {
    CREATE: API_ROOT + '/smart_parking/bookings/',
    DETAIL: (id) => API_ROOT + `/smart_parking/bookings/${id}/`,
    ACTIVE_SESSION: API_ROOT + '/smart_parking/bookings/active_session/',
    HISTORY: (page) =>
      API_ROOT + `/smart_parking/bookings/history/?page=${page}`,
    SCAN_TO_BOOK: API_ROOT + '/smart_parking/bookings/scan_to_book/',
    SCAN_TO_CONFIRM: API_ROOT + '/smart_parking/bookings/scan_to_confirm/',
    EXTEND_INFO: (id) =>
      API_ROOT + `/smart_parking/bookings/${id}/extend_info/`,
    EXTEND: (id) => API_ROOT + `/smart_parking/bookings/${id}/extend/`,
    CANCEL: (id) => API_ROOT + `/smart_parking/bookings/${id}/cancel/`,
    STOP: (id) => API_ROOT + `/smart_parking/bookings/${id}/stop/`,
  },
  EMERGENCY: {
    CONTACTS: API_ROOT + '/emergency_button/contacts/',
  },
  BILLING: {
    LIST_PAYMENT_METHODS_BY_COUNTRY: (code) =>
      API_ROOT + `/billing/list-methods/${code}/`,
    PAYMENT: {
      STRIPE: {
        PROCESS: (id) => API_ROOT + `/billing/payments/stripe/process/${id}/`,
        ADD_CARD: API_ROOT + '/billing/payments/stripe/add_card/',
        CREATE_PAYMENT_INTENT: (id) =>
          API_ROOT + `/billing/payments/stripe/sca/create-payment/${id}/`,
        PAYMENT_INTENT_SUCCESS: (intent_id) =>
          API_ROOT +
          `/billing/payments/stripe/sca/payment_success/${intent_id}/`,
      },
    },
    DEFAULT_PAYMENT_METHODS: API_ROOT + '/billing/default_payment_method/',
  },
  NOTIFICATION: {
    REGISTER_SIGNAL_ID: API_ROOT + '/accounts/register-signal-id/',
    LIST_ALL_NOTIFICATIONS: (page, type) =>
      API_ROOT + `/notifications/notifications/?page=${page}&type=${type}`,
    SET_READ: (id) => API_ROOT + `/notifications/notifications/${id}/set_read/`,
    NUMBER: API_ROOT + '/notifications/notifications/number/',
  },
  EMERGENCY_BUTTON: {
    CREATE_CONTACT: API_ROOT + '/emergency_button/contacts/',
    CONTACTS: API_ROOT + '/emergency_button/contacts/',
    REMOVE_CONTACTS: (id) => API_ROOT + `/emergency_button/contacts/${id}/`,
    SEND_ALERT: API_ROOT + '/emergency_button/events/',
    NOTIFY_SMS: (id) => API_ROOT + `/emergency_button/events/${id}/notify_sms/`,
    FASLE_ALARM: (id) =>
      API_ROOT + `/emergency_button/events/${id}/false_alarm/`,
    RESOLVE: (id) => API_ROOT + `/emergency_button/events/${id}/resolve/`,
  },
  IOT: {
    CHIP_MANAGER: {
      WATCH_CONFIGS: API_ROOT + '/chip_manager/watch_configs/',
      PUSHER_AUTH: API_ROOT + '/chip_manager/pusher/auth/',
    },
  },
};

export default API;
