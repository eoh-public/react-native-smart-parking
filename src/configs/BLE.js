export default {
  BLE_REMOTE_SERVICE_UUID: '4fafc201-1fb5-459e-8fcc-c5c9c331914b',
  BLE_REMOTE_CHARACTERISTIC_UUID: 'beb5483e-36e1-4688-b7f5-ea07361b26a8',
};
