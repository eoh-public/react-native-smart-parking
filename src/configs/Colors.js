export const Colors = {
  Primary: '#00979D',
  Text: '#363636',
  Golden: '#b2973f',
  White: '#FFFFFF',
  Red: 'red',
  Green: 'green',
  Gray: '#838383',
  LightGray: '#989898',
  Error: '#F10303',
  DarkLayer: '#00000080',
  LightDarkLayer: '#0000004C',
  LightingDarkLayer: '#00000033',
  LightWhiteLayer: '#F7F7F7D9',
  CoinTagBG: '#F7F7F7',
  Black: '#000000',
  Disabled: '#f8f8f8',
  Border: '#BFBFBF',
  ButtonTag: '#989898',
  WrapGray: '#F5F5F5',
  SwitchButton: '#909090',
  Read: '#F3F3F3',
  Orange: '#F08229',
  BG: '#F0F2F5',
  BGImage: '#F4F8F8',
  Transparent: '#00000000',
  TransparentWhite: '#FFFFFF00',
  BGMapCircle: '#E6F7FF30',
  BGNotRain: '#FEFFE6',
  BorderRain: '#36CFC9',
  Shadow: '#303133',
  ShadownTransparent: 'rgba(48, 49, 51, 0.1)',
  TextTransparent: 'transparent',
  TextGray: 'gray',
  LightOrange: '#FEEFE7',

  //Range Cyan
  Cyan2: '#B5F5EC',
  Cyan2opa30: '#B5F5EC4D',
  Cyan6: '#13C2C2',

  //Range GeekBlue
  GeekBlue3: '#ADC6FF',

  // Range Gray
  Gray4: '#E8E8E8',
  Gray5: '#D9D9D9',
  Gray6: '#BFBFBF',
  Gray7: '#8C8C8C',
  Gray8: '#595959',
  Gray9: '#262626',
  Gray2: '#FAFAFA',
  Gray3: '#F5F5F5',
  Gray11: '#3031331a',
  Gray12: '#DDD',
  Gray13: '#CCC',
  Gray14: '#EEE',
  Gray16: '#333',
  Gray17: '#D3D3D3',
  Gray18: '#9B9A9B',
  Gray19: '#E5E5E5',

  // Range Green
  Green1: '#F6FFED',
  Green3: '#B7EB8F',
  Green6: '#52C41A',
  Green7: '#00979D',
  Green8: '#D9F7BE',
  Green9: '#389E0D',
  Green10: '#237804',

  //Yellow
  Yellow6: '#FADB14',
  Yellow4: '#FFF566',

  //Orange
  Orange3: '#FFD591',
  Orange4: '#FFC069',
  Orange6: '#FA8C16',
  Orange7: '#FAAD14',

  //Red
  Red9: '#A8071A',
  Red8: '#CF1322',
  Red7: '#FA541C',
  Red6: '#F5222D',
  Red5: '#FF4D4F',
  Red4: '#FF7875',
  Red3: '#FFA39E',
  Red2: '#FFCCC7',
  Red1: '#FFF1F0',

  //Purple
  Purple3: '#D3ADF7',
  Purple6: '#722ED1',

  //Pink
  Pink1: '#EFDBFF',

  //Blue
  Blue1: '#F0F5FF',
  Blue2: '#D6E4FF',
  Blue3: '#ADC6FF',
  Blue4: '#85A5FF',
  Blue5: '#597EF7',
  Blue6: '#2F54EB',
  Blue7: '#1D39C4',
  Blue9: '#91D5FF',
  Blue10: '#1890FF',
  Blue11: '#096DD9',
  Blue12: '#0050B3',

  //Range Volcano:
  Volcano3: '#FFBB96',

  //lime
  Lime6: '#A0D911',

  Cyan1: '#E6FFFB',

  LightPrimary: '#EBF7F8',

  BlackTransparent1: 'rgba(0, 0, 0, 0.1)',
  BlackTransparent2: 'rgba(0, 0, 0, 0.2)',
  BlackTransparent3: 'rgba(0, 0, 0, 0.3)',
  BlackTransparent4: 'rgba(0, 0, 0, 0.4)',
  BlackTransparent5: 'rgba(0, 0, 0, 0.5)',
  BlackTransparent6: 'rgba(0, 0, 0, 0.6)',
  BlackTransparent16: 'rgba(0, 0, 0, 0.16)',
  WhiteTransparent8: 'rgba(255, 255, 255, 0.8)',
};
