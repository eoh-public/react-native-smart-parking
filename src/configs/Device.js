/** @format */

import { Dimensions, PixelRatio, Platform } from 'react-native';

const { width, height } = Dimensions.get('window');

const isIphoneX =
  Platform.OS === 'ios' &&
  !Platform.isPad &&
  !Platform.isTVOS &&
  (height >= 812 || width >= 812);

export default {
  pixelRatio: PixelRatio.get(),
  screenWidth: width,
  screenHeight: height,
  isIOS: Platform.OS === 'ios',
  isIphoneX,
  ToolbarHeight: isIphoneX ? 44 : Platform.OS === 'ios' ? 20 : 0,
  TopbarHeight: isIphoneX ? 88 : Platform.OS === 'ios' ? 64 : 44, //incule toolbar
  androidSmallHeight: Platform.OS === 'android' && height < 600,
};
