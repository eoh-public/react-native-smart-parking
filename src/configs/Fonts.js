export const Fonts = {
  Regular: 'SFProDisplay-Regular',
};
