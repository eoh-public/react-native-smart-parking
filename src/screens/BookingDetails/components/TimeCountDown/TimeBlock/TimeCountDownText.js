import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';

import Text from '../../../../../commons/Text';
import { Colors } from '../../../../../configs';
import { TESTID } from '../../../../../configs/Constants';

const TimeCountDownText = memo(({ number, color }) => {
  return (
    <View style={styles.container}>
      <View style={styles.topColor} />
      <View style={styles.bottomColor} />
      <Text
        testID={TESTID.TIME_COUNT_DOWN_TEXT}
        type="H2"
        style={styles.textNumber}
        color={color}
        semibold
      >
        {number}
      </Text>
    </View>
  );
});

export default TimeCountDownText;

const styles = StyleSheet.create({
  container: {
    width: 32,
    height: 48,
    borderRadius: 2,
    marginRight: 4,
    shadowColor: Colors.Black,
    shadowOffset: {
      width: 2,
      height: 3,
    },
    shadowOpacity: 0.1,
    shadowRadius: 2,

    elevation: 1,
    alignItems: 'center',
  },
  topColor: {
    position: 'absolute',
    width: 32,
    height: 24,
    backgroundColor: Colors.Gray3,
    borderTopLeftRadius: 2,
    borderTopRightRadius: 2,
  },
  bottomColor: {
    width: 32,
    height: 24,
    position: 'absolute',
    top: 24,
    backgroundColor: Colors.Gray4,
    borderBottomLeftRadius: 2,
    borderBottomRightRadius: 2,
  },
  textNumber: {
    marginTop: 8,
    textAlign: 'center',
    width: 32,
  },
});
