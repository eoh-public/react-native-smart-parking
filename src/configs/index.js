import { SPConfig, initSPConfig } from './SPConfig';
import API from './API';
import { Colors } from './Colors';
import { Constants } from './Constants';
import Theme from './Theme';
import Images from './Images';
import Styles from './Styles';
import BLE from './BLE';
import Device from './Device';
import { Fonts } from './Fonts';
export {
  API,
  BLE,
  Colors,
  Constants,
  Theme,
  Styles,
  Images,
  Device,
  Fonts,
  SPConfig,
  initSPConfig,
};
