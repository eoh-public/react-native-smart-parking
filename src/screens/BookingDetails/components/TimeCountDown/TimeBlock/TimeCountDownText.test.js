import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { Colors } from '../../../../../configs';
import TimeCountDownText from './TimeCountDownText';

describe('Test TimeCountDownText', () => {
  let tree;
  test('render TimeCountDownText', () => {
    act(() => {
      tree = renderer.create(
        <TimeCountDownText number={'02'} color={Colors.Red} />
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
