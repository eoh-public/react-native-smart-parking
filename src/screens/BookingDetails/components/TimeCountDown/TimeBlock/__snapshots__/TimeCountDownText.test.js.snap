// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TimeCountDownText render TimeCountDownText 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 2,
      "elevation": 1,
      "height": 48,
      "marginRight": 4,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 3,
        "width": 2,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
      "width": 32,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "borderTopLeftRadius": 2,
        "borderTopRightRadius": 2,
        "height": 24,
        "position": "absolute",
        "width": 32,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "borderBottomLeftRadius": 2,
        "borderBottomRightRadius": 2,
        "height": 24,
        "position": "absolute",
        "top": 24,
        "width": 32,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "red",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 24,
          "lineHeight": 32,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginTop": 8,
          "textAlign": "center",
          "width": 32,
        },
      ]
    }
    testID="TIME_COUNT_DOWN_TEXT"
  >
    02
  </Text>
</View>
`;
